psydac/api/postprocessing.py

```diff
@@ -79,3 +88,759 @@ def get_grid_lines_2d(domain_h, V_h, *, refine=1):
+
+    def _add_vector_space(self, vector_space, name=None):
+        """Add a vector space to the scope of this instance of OutputManager.
+
+        Parameters
+        ----------
+        vector_space: psydac.fem.vector.VectorFemSpace
+            Vector FemSpace to add to the scope.
+        """
+
+        symbolic_space = vector_space.symbolic_space
+        dim = symbolic_space.domain.dim
+        patch_name = symbolic_space.domain.name
+        kind = symbolic_space.kind
+
+        scalar_spaces_info = []
+        for i, scalar_space in enumerate(vector_space.spaces):
+            sc_space_info = self._add_scalar_space(scalar_space,
+                                                   name=name + f'[{i}]',
+                                                   dim=dim,
+                                                   patch_name=patch_name,
+                                                   kind='UndefinedSpaceType()')
+            scalar_spaces_info.append(sc_space_info)
+
+        spaces_info = self._spaces_info
+
+        new_vector_space = {'name': name,
+                            'kind': self.space_types_to_str[str(kind)],
+                            'components': scalar_spaces_info,
+                            }
+
+        patch_index = [spaces_info['patches'][i]['name']
+                       for i in range(len(spaces_info['patches']))].index(patch_name)
+
+        try:
+            spaces_info['patches'][patch_index]['vector_spaces'].append(new_vector_space)
+        except KeyError:
+            spaces_info['patches'][patch_index].update({'vector_spaces': [new_vector_space]})
+
+        self._spaces_info = spaces_info
+        self._spaces.append(vector_space)
+        self._spaces.append(name)
+
+    def export_fields(self, **fields):
+        """
+        A function that exports the fields' coefficients to an HDF5 file. They are
+        saved under a snapshot/patch/space/field scheme or static/patch/space/field.
+
+        If the saving scheme is time dependent, unsets it to avoid writing twice to
+        the same snapshot.
+
+        Parameters
+        ----------
+        fields : psydac.fem.basic.FemField dict
+            List of named fields
+
+        Raises
+        ------
+        ValueError
+            When self.is_static is None (no saving scheme specified)
+        """
+        if self.is_static is None:
+            raise ValueError('Saving scheme not specified')
+
+        # For now, I assume that if something is mpi parallel everything is
+        space_test = list(fields.values())[0].space
+        if space_test.is_product:
+            comm = space_test.spaces[0].vector_space.cart.comm if space_test.vector_space.parallel else None
+        else:
+            comm = space_test.vector_space.cart.comm if space_test.vector_space.parallel else None
+
+        # Open HDF5 file (in parallel mode if MPI communicator size > 1)
+        kwargs = {}
+        if comm is not None:
+            if comm.size > 1:
```

Comment:
In classical Pythonic fashion, thanks to short-circuiting of the boolean operators, you can just write
```python
if comm and comm.size > 1:
```
because in the case where `comm is None` the Python interpreter knows that the logical expression is `False` and stops before evaluating `comm.size > 1`.

Reply:
Fixed in 0c5bd9b